src/bidiMapper/domains/session/EventManager.ts

```diff
@@ -188,17 +192,31 @@ export class EventManager extends EventEmitter<EventManagerEventsMap> {
         }
       }
     }
+
+    await this.toggleModulesIfNeeded();
   }
 
-  unsubscribe(
+  async unsubscribe(
     eventNames: ChromiumBidi.EventNames[],
     contextIds: (BrowsingContext.BrowsingContext | null)[],
     channel: BidiPlusChannel
-  ) {
+  ): Promise<void> {
     for (const name of eventNames) {
       assertSupportedEvent(name);
     }
     this.#subscriptionManager.unsubscribeAll(eventNames, contextIds, channel);
+    await this.toggleModulesIfNeeded();
+  }
+
+  async toggleModulesIfNeeded(): Promise<void> {
+    // TODO: Only update changed subscribers
+    await Promise.all(
```

Comment:
Go over all targets or or all browsing contexts instead of only top-level contexts